### PR TITLE
Service now v2 bug

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: demisto/etc#32208

## Description
Added the arguments` reassignment_count, reopen_count, sys_updated_by, sys_updated_on` to the commands `servicenow-update-ticket` and `servicenow-create-ticket`, plus adjusted that all arguments that are modified manually by the user using these commands are now inserted to context data automatically. 
The argumnets `display` and `escalation` were removed from `servicenow-update-ticket` and `servicenow-create-ticket`.

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by Noy-Maimon** [comment]
@roysagi Hi, can you please take a look at the RN? thanks

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -969,6 +970,13 @@ def update_ticket_command(client: Client, args: dict) -> Tuple[Any, Dict, Dict,
     hr_ = get_ticket_human_readable(ticket, ticket_type, additional_fields_keys)
     human_readable = tableToMarkdown(f'ServiceNow ticket updated successfully\nTicket type: {ticket_type}',
                                      t=hr_, removeNull=True)
+
+    # make the modified fields the user inserted as arguments show in the context
+    if additional_fields:
+        additional_fields_keys.extend(list(args.keys()))
+    else:
+        additional_fields_keys = list(args.keys())
```
Comment (bakatzir) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:978]: can there be duplicate keys?

File: Packs/ServiceNow/ReleaseNotes/2_1_4.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (bakatzir) [inline Packs/ServiceNow/ReleaseNotes/2_1_4.md:1]: please add an entry for the script update of the DO

Comment (bakatzir) [reply Packs/ServiceNow/ReleaseNotes/2_1_4.md:1]: also mention you removed the display and escalation args as they did not ever work

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 3]
```diff
@@ -80,12 +80,12 @@ def test_split_fields():
```
Comment (bakatzir) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:81]: why did not you add a test case for the `get_ticket_context` additional_keys+args addiiton?

Comment (Noy-Maimon) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:81]: those are being tested by the modification i did in the UT 'test_command'. I can add a designated one if needed but the current one is already testing that.

File: Packs/ServiceNow/ReleaseNotes/2_1_4.md (hunk 2 above)
Comment (Noy-Maimon) [reply Packs/ServiceNow/ReleaseNotes/2_1_4.md:1]: done

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py (hunk 3 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:81]: I would still suggest adding a separate test case. i'll write the docstring to explain:
given - additional keys of a ticket alongside regular keys.
when - updating a ticket
then - validate that all the details of the ticket were updated, and all the updated keys are showen in the HR and in the context.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 1 above)
Comment (bakatzir) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:978]: not. but would still: list(set(a).union(set(b))
why leave this edge open?
also a good 2nd test case

Comment (Noy-Maimon) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py]: done

File: Packs/ServiceNow/ReleaseNotes/2_1_4.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+- Manually changed ticket fields, done by the suitable arguments in ***servicenow-update-ticket*** and
+ ***servicenow-create-ticket***, are now inserted automatically to context data.
+- Updated docker image to: demisto/python3:3.9.1.14969.
+
+##### ServiceNow_CMDB
+- Updated docker image to: demisto/python3:3.9.1.14969.
+
+#### Scripts
+##### ServiceNowIncidentStatus
+- Updated docker image to: demisto/python3:3.9.1.14969.
```
Comment (bakatzir) [inline Packs/ServiceNow/ReleaseNotes/2_1_4.md:17]:
```suggestion
#### Integrations
##### ServiceNow v2
- Added to *the **servicenow-update-ticket*** and ***servicenow-create-ticket*** commands the arguments: *reassignment_count*, *reopen_count*, *sys_updated_by* and *sys_updated_on*.
- Removed the arguments *display* and *escalation* from the ***servicenow-update-ticket*** and ***servicenow-create-ticket*** commands, as they are not supporterd in the API.
- Added the mentioned ticket fields in thearguments given to the ***servicenow-update-ticket*** and ***servicenow-create-ticket*** commands to the context data.
- Updated the Docker image to: *demisto/python3:3.9.1.14969*.

##### ServiceNow_CMDB
- Updated the Docker image to: *demisto/python3:3.9.1.14969*.

#### Scripts
##### ServiceNowIncidentStatus
- Updated the Docker image to: *demisto/python3:3.9.1.14969*.
```

File: Packs/ServiceNow/ReleaseNotes/2_1_6.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+
+#### Integrations
+##### ServiceNow v2
+- Added to *the **servicenow-update-ticket*** and ***servicenow-create-ticket*** commands the arguments: *reassignment_count*, *reopen_count*, *sys_updated_by* and *sys_updated_on*.
```
Comment (bakatzir) [inline Packs/ServiceNow/ReleaseNotes/2_1_6.md:4]:
```suggestion
- Added to the **servicenow-update-ticket*** and ***servicenow-create-ticket*** commands the arguments: *reassignment_count*, *reopen_count*, *sys_updated_by* and *sys_updated_on*.
```

File: Packs/ServiceNow/ReleaseNotes/2_1_6.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+- Removed the arguments *display* and *escalation* from the ***servicenow-update-ticket*** and ***servicenow-create-ticket*** commands, as they are not supporterd in the API.
```
Comment (bakatzir) [inline Packs/ServiceNow/ReleaseNotes/2_1_6.md:5]:
```suggestion
- Removed the arguments *display* and *escalation* from the ***servicenow-update-ticket*** and ***servicenow-create-ticket*** commands, as they are not supportקd in the API.
```

**Review by bakatzir — APPROVED** [review]

**Pull request merged**